### ISSUE TITLE
Clarification about the mask

### ISSUE THREAD

**Issue by FrancescoSaverioZuppichini** [microsoft/Swin-Transformer#125]
Hi guys, 

I have trouble understanding what is going on here

```python
 # calculate attention mask for SW-MSA
            H, W = self.input_resolution
            img_mask = torch.zeros((1, H, W, 1))  # 1 H W 1
            h_slices = (slice(0, -self.window_size),
                        slice(-self.window_size, -self.shift_size),
                        slice(-self.shift_size, None))
            w_slices = (slice(0, -self.window_size),
                        slice(-self.window_size, -self.shift_size),
                        slice(-self.shift_size, None))
            cnt = 0
            for h in h_slices:
                for w in w_slices:
                    img_mask[:, h, w, :] = cnt
                    cnt += 1
```

I don't really understand why we need `cnt`. 

Thanks!

Cheers,

Francesco

**Comment by zeliu98** [comment]
Hi @FrancescoSaverioZuppichini, the **cnt** is used to distinguish different areas. For points in different area, we need to add a attention mask for them. 
You can find a discussion in this issue: https://github.com/microsoft/Swin-Transformer/issues/38#issuecomment-823806591
